File: engine/src/main/java/com/terralite/engine/physics/Transform.java
package com.terralite.engine.physics;

public record Transform(double x, double y, double z) {
    public static final Transform ORIGIN = new Transform(0.0, 0.0, 0.0);

    public Transform translate(double dx, double dy, double dz) {
        return new Transform(x + dx, y + dy, z + dz);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/physics/Transform.kt
package com.terralite.engine.physics

@JvmRecord
data class Transform(val x: Double, val y: Double, val z: Double) {
    companion object {
        @JvmField val ORIGIN = Transform(0.0, 0.0, 0.0)
    }

    fun translate(dx: Double, dy: Double, dz: Double): Transform =
        Transform(x + dx, y + dy, z + dz)
}
